object.c

```diff
@@ -3651,15 +3651,37 @@ rb_String(VALUE val)
 
 /*
  *  call-seq:
- *     String(arg)   -> string
+ *    String(object) -> object or new_string
  *
- *  Returns <i>arg</i> as a String.
+ *  Returns an array converted from +object+.
  *
- *  First tries to call its <code>to_str</code> method, then its <code>to_s</code> method.
+ *  If +object+ is:
+ *
+ *  - A string, returns +object+.
+ *  - +nil+, returns an empty string.
+ *
+ *  Otherwise, if +object+ responds to:
+ *
+ *  - +:to_str+, returns <tt>object.to_str</tt>.
+ *  - +:to_s+, returns <tt>object.to_s</tt>.
```

Comment:
These only work if `to_str` or `to_s` return a string.  If they return another value, TypeError is raised.  Similar for array and hash.

Reply:
Fixed.